site/content/integrate/apps/quick-start-js/_index.md

```diff
@@ -0,0 +1,301 @@
+                        call: {
+                            path: "/send-modal"
+                        }
+                    }
+                ]
+            },
+            {
+                location: "/command",
+                bindings: [
+                    {
+                        icon: "http://localhost:8080/static/icon.png",
+                        label: "helloworld",
+                        description: "Hello World app",
+                        hint: "[send]",
+                        bindings: [
+                            {
+                                location: "send",
+                                label: "send",
+                                call: {
+                                    path: "/send"
+                                }
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    }));
+    break;
+```
+
+### Functions and form
+
+Functions handle user events and webhooks. The Hello World app exposes two functions:
+
+- `/send` that services the command and modal.
+- `/send-modal` that forces the modal to be displayed.
+
+The functions use a simple form with one text field named `"message"`, the form submits to `/send`.
+
+`/send/form` and  `send-modal/submit` both return the same data:
+
+```js
+case "/send/form":
+case "/send-modal/submit":
+    res.setHeader("Content-Type", "application/json");
+    res.end(JSON.stringify({
+        type: "form",
+        form: {
+            title: "Hello, world!",
+            icon: "http://localhost:8080/static/icon.png",
+            fields: [
+                {
+                    type: "text",
+                    name: "message",
+                    label: "message"
+                }
+            ],
+            call: {
+                path: "/send"
+            }
+        }
+    }));
+    break;
+```
+
+## Icons 
+
+Apps may include static assets. One example that was already used above is the `icon` for the two bindings. Static assets must be served under the `static` path.
+
+Download an example icon using:
+
+```bash
+wget https://github.com/mattermost/mattermost-plugin-apps/raw/master/examples/go/helloworld/icon.png
+```
+
+`fs` is used to load the file:
+
+```js
+const fs = require("fs").promises;
+```
+
+And then add another `case` to handle the icon:
+```js
+case "/static/icon.png":
+    fs.readFile(__dirname + "/icon.png")
+        .then(contents => {
+            res.setHeader("Content-Type", "image/png");
+            res.end(contents);
+        })
+    break;
+```
+
+### Serving the data
+
+Finally, add the application logic that gets executed when either the slash command is run or the modal submitted:
+
+```js
+case "/send/submit":
+    let body = "";
+    req.on("data", chunk => {
+        body += chunk;
+    });
+    req.on("end", () => {
+        const call = JSON.parse(body);
+
+        let message = "Hello, world!";
```

Comment:
I suggest using an express server instead of going through this trouble. It's as simple as `npm i express` to have all the necessary dependencies. A developer will be using express when they develop their app anyway, or another http framework that they are already familiar with.